### PR TITLE
Fixed bug for system crashing  if back arrow is clicked  after printing or correcting a  record

### PR DESCRIPTION
**Pull request by favour234**
- added function to  replace history to the present window's location after correcting a record or printing a certificate to prevent the browser's back button from going back.
- updated isLandingPage method.

### CODE REVIEW

File: packages/client/src/views/OfficeHome/readyForReview/ReadyForReview.tsx [hunk 1]
```diff
@@ -101,10 +101,17 @@ class ReadyForReviewComponent extends React.Component<
...
+
     window.addEventListener('resize', this.recordWindowWidth)
   }
 
+  popStateListener() {
+    window.history.pushState(null, document.title, window.location.href)
+  }
+
   componentWillUnmount() {
+    window.addEventListener('popstate', this.popStateListener)
```
Comment (euanmillar) [inline packages/client/src/views/OfficeHome/readyForReview/ReadyForReview.tsx:114]: @favour234 I dont like this solution.  You add a listener on mount and unmount.  What is wrong with how you approached https://github.com/opencrvs/opencrvs-core/pull/4050 ?

Why are you removing code in Header.tsx that you added in that PR? ``` (REGISTRAR_ROLES.includes(role as string) &&
        location.includes(REGISTRAR_HOME))```

**Review by euanmillar — CHANGES_REQUESTED** [review]
@favour234 I am not convinced by the solution you have coded and how it may affect a previous bugfix https://github.com/opencrvs/opencrvs-core/pull/4050

File: packages/client/src/views/OfficeHome/readyForReview/ReadyForReview.tsx (hunk 1 above)
Comment (favour234) [reply packages/client/src/views/OfficeHome/readyForReview/ReadyForReview.tsx:114]: the code "(REGISTRAR_ROLES.includes(role as string) && location.includes(REGISTRAR_HOME))" resolved the issue but unknowingly created another which @CornelMaze  worked on.

Comment (favour234) [reply packages/client/src/views/OfficeHome/readyForReview/ReadyForReview.tsx]: > @favour234 I dont like this solution. You add a listener on mount and unmount. What is wrong with how you approached #4050 ?
> 
> Why are you removing code in Header.tsx that you added in that PR? ` (REGISTRAR_ROLES.includes(role as string) && location.includes(REGISTRAR_HOME))`



> @favour234 I dont like this solution. You add a listener on mount and unmount. What is wrong with how you approached #4050 ?
> this code was also added in my previous pr https://github.com/opencrvs/opencrvs-core/pull/4050/files.
After redirecting to home page on correction or printing of a certificate, its observed that we still had access to go back from chromes  browser back button, which result to system crashing , So the popStateListener  is basically to change the history to the present state, which is the present location of the page..so if you click on the back button on the browser ..you still remain on that page.  its a little function to replace  windows history to the present window's location.
 
 
> Why are you removing code in Header.tsx that you added in that PR? ` (REGISTRAR_ROLES.includes(role as string) && location.includes(REGISTRAR_HOME))`

**Comment by favour234** [comment]
> @favour234 I am not convinced by the solution you have coded and how it may affect a previous bugfix #4050

this code was also used in my previous pr https://github.com/opencrvs/opencrvs-core/pull/4050/files. the approach on this pr was handling for the arrow back button on the system not the browser., so after redirecting to home page on correction or printing of a certificate, we still had access to go back from chromes browser back button, which result to system crashing , So the popStateListener is basically to change the history to the present state, which is the present location of the page..so if you click on the back button on the browser, after making a correction or printing a certificate, it restricts you from accessing the previous page . It also covers for the system arrow back button as well.

**Comment by favour234** [comment]
> @favour234 I am not convinced by the solution you have coded and how it may affect a previous bugfix #4050

this code was also used in my previous pr https://github.com/opencrvs/opencrvs-core/pull/4050/files. the approach on this pr was handling for the arrow back button on the system not the browser., so after redirecting to home page on correction or printing of a certificate, we still had access to go back from chromes browser back button, which result to system crashing , So the popStateListener is basically to change the history to the present state, which is the present location of the page..so if you click on the back button on the browser, after making a correction or printing a certificate, it restricts you from accessing the previous page . It also covers for the system arrow back button as well.

File: packages/client/src/components/Header/Header.tsx [hunk 2]
```diff
@@ -361,13 +361,13 @@ class HeaderComp extends React.Component<IFullProps, IState> {
...
+      (FIELD_AGENT_ROLES.includes(role as string) && HOME.includes(location)) ||
       (NATL_ADMIN_ROLES.includes(role as string) &&
-        location.includes(PERFORMANCE_HOME)) ||
+        PERFORMANCE_HOME.includes(location)) ||
       (SYS_ADMIN_ROLES.includes(role as string) &&
-        location.includes(PERFORMANCE_HOME)) ||
+        PERFORMANCE_HOME.includes(location)) ||
       (REGISTRAR_ROLES.includes(role as string) &&
-        location.includes(REGISTRAR_HOME))
+        REGISTRAR_HOME.includes(location))
```
Comment (Zangetsu101) [inline packages/client/src/components/Header/Header.tsx:370]: These changes will cause this bug to reappear https://github.com/opencrvs/opencrvs-core/issues/4203 so we should not do this

Comment (favour234) [reply packages/client/src/components/Header/Header.tsx:370]: okay

File: packages/client/src/views/OfficeHome/readyForReview/ReadyForReview.tsx (hunk 1 above)
Comment (Zangetsu101) [reply packages/client/src/views/OfficeHome/readyForReview/ReadyForReview.tsx:114]: @favour234 You need `removeEventListener` in unmount and not add

File: packages/client/src/components/Header/Header.tsx [hunk 3]
```diff
@@ -365,9 +365,7 @@ class HeaderComp extends React.Component<IFullProps, IState> {
       (NATL_ADMIN_ROLES.includes(role as string) &&
         location.includes(PERFORMANCE_HOME)) ||
       (SYS_ADMIN_ROLES.includes(role as string) &&
-        location.includes(PERFORMANCE_HOME)) ||
-      (REGISTRAR_ROLES.includes(role as string) &&
-        location.includes(REGISTRAR_HOME))
```
Comment (Zangetsu101) [inline packages/client/src/components/Header/Header.tsx:370]: Is this change necessary? Because it might again cause issues with the history buttons

**Review by Zangetsu101 — APPROVED** [review]

**Review by Zangetsu101 — CHANGES_REQUESTED** [review]

**Review by Zangetsu101 — APPROVED** [review]
